userspace/libsinsp/sinsp.h

```diff
@@ -1121,7 +1122,8 @@ VISIBILITY_PRIVATE
 	bool m_is_dumping;
 	const scap_machine_info* m_machine_info;
 	const scap_agent_info* m_agent_info;
-	scap_stats_v2 m_sinsp_stats_v2[SINSP_MAX_RESOURCE_UTILIZATION];
+	sinsp_stats_v2 m_sinsp_stats_v2;
```

Comment:
Since stats were protected by a macro before, can we retains a similar approach by making this back a `unique_ptr` and only creating it if requested (ie: by a `sinsp_enable_stats()` API or similar that must be called before the `open`ing of the inspector).

Reply:
Maybe we don't need a unique_ptr, just ensure the struct is zero-initialized and have an option to disable the stats updates?

If you disable the stats and then read them and expect non-zero values, I'm not sure you have someone else to blame :D

Reply:
This would save us lots of `if m_sinsp_stats_v2 != nil` checks, neat!

Reply:
@gnosek ^ you also onboard, if yes can go ahead and re-push later.

Reply:
@FedeDP pushed a suggestion, plz let me know. It seems checking the inspector would only be needed in one place and in fact in most cases it's already de-referenced anyways before.

In addition, luckily had a pod sandbox running in the background and realized we should not be counting sandboxes as missing container images ... that's the one important metric I really would like to check on in prod, so we better get it right.